api/server/router/system/system_routes.go

```diff
@@ -143,6 +145,40 @@ func (s *systemRouter) getEvents(ctx context.Context, w http.ResponseWriter, r *
 	}
 }
 
+func getEventsFilters(filtersParam, version string) (filters.Args, error) {
+	ef, err := filters.FromParam(filtersParam)
+	if err != nil {
+		return filters.NewArgs(), err
+	}
+	if versions.LessThan(version, "1.25") {
+		// #25798: handle old event filtering for exec_start, exec_create and health_status
+		if ef.Include("event") {
+			for _, ev := range ef.Get("event") {
+				// health_status
+				convertEventFilter(ef, ev, "health_status", "state")
+				// exec_create
+				convertEventFilter(ef, ev, "exec_start", "command")
+				// exec_start
+				convertEventFilter(ef, ev, "exec_start", "command")
+			}
+		}
+	}
+	return ef, nil
+}
+
+func convertEventFilter(ef filters.Args, event, name, attribute string) {
+	if strings.HasPrefix(event, name) {
+		ef.Del("event", event)
+		ef.Add("event", name)
+		values := strings.SplitN(event, ":", 2)
+		if len(values) == 2 {
+			value := strings.TrimSpace(values[1])
+			ef.Add("label", fmt.Sprintf("%s=%s", attribute, value))
```

Comment:
This is a little bit hacky as `labels` are, in the events, store at the same level as `attributes`, so we can filter on any additionnal attributes using the label filter.